### PR TITLE
Hide modules whose apis have not yet been finalised for 0.7.0

### PR DESCRIPTION
**Pull request by pranaysashank**
Updated Changelog, the benchmarking ops, and the File IO example in README as well

### CODE REVIEW

File: examples/FileIOExamples.hs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import qualified Streamly.Prelude as S
 import qualified Streamly.Fold as FL
 import qualified Streamly.Mem.Array as A
-import qualified Streamly.FileSystem.File as File
+import qualified Streamly.FileSystem.Handle as Handle
```
Comment (harendra-kumar) [inline examples/FileIOExamples.hs]: Why this change? Is this accidental?

Comment (pranaysashank) [reply examples/FileIOExamples.hs]: Yes, this is accidental.

File: examples/FileSinkServer.hs [hunk 2]
```diff
@@ -9,21 +9,24 @@ import System.Environment (getArgs)
...
 import qualified Streamly.Network.Server as NS
 import qualified Streamly.Prelude as S
 
+import System.IO (withFile, IOMode(..))
+
 main :: IO ()
 main = do
     file <- fmap head getArgs
-    File.append file
+    withFile file AppendMode
```
Comment (harendra-kumar) [inline examples/FileSinkServer.hs]: Where is `withFile` coming from?

Comment (pranaysashank) [reply examples/FileSinkServer.hs]: From System.IO

Comment (harendra-kumar) [reply examples/FileSinkServer.hs]: Now I see, its right there, my brain just filtered that out because I was expecting it to come from some Streamly module.

**Comment by harendra-kumar** [comment]
This change will render some APIs unusable but visible, because types like `AbsTime` are not available. Should we hide them instead, users will think that they are there but they can't use them?

**Comment by pranaysashank** [comment]
Oh yeah, I forgot about the apis that depend on time. Those apis are cool, I guess we can either use a `Num a` instead of AbsTime (which would make the change to AbsTime a breaking change later) or hide them and ship them later

**Comment by harendra-kumar** [comment]
Let's hide them for now, they can be added later in a minor release as well.
